src/signature/commitment_signature.rs

```diff
@@ -0,0 +1,292 @@
+// Copyright 2021 The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+//! Digital Signature module
+//! This module defines generic traits for handling the digital signature operations, agnostic
+//! of the underlying elliptic curve implementation
+
+use crate::{
+    commitment::{HomomorphicCommitment, HomomorphicCommitmentFactory},
+    keys::{PublicKey, SecretKey},
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    cmp::Ordering,
+    hash::{Hash, Hasher},
+    marker::PhantomData,
+    ops::{Add, Mul},
+};
+use tari_utilities::{hex::Hex, ByteArray};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Error, PartialEq, Eq, Deserialize, Serialize)]
+pub enum CommitmentSignatureError {
+    #[error("An invalid challenge was provided")]
+    InvalidChallenge,
+}
+
+#[allow(non_snake_case)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CommitmentSignature<P, K, C> {
+    public_commitment_nonce: HomomorphicCommitment<P>,
+    signature_u: K,
```

Comment:
I don't know if "signature" adds any value to the name here
```suggestion
    u: K,
```

Reply:
trying not to use single char names, but did update to use `u` and `v` rather